src/s3.rs

```diff
@@ -0,0 +1,211 @@
+            .bucket(bucket)
+            .key(key)
+            .body(ByteStream::from(data))
+            .send()
+            .await;
+
+        match result {
+            Ok(_) => Ok(()),
+            Err(e) => {
+                let error_message =
+                    format!("error while s3 put_object {}/{}; err={}", bucket, key, e);
+                tracing::error!(error_message);
+                Err(AwsError::new(&*error_message))
+            }
+        }
+    }
+
+    pub async fn get_object(&self, bucket: &str, key: &str) -> Result<ByteStream, AwsError> {
+        let result = self
+            .client
+            .get_object()
+            .bucket(bucket)
+            .key(key)
+            .send()
+            .await;
+        match result {
+            Ok(object) => Ok(object.body),
+            Err(e) => {
+                let error_message =
+                    format!("error while s3 get_object {}/{}; err={}", bucket, key, e);
+                tracing::error!(error_message);
+                Err(AwsError::new(&*error_message))
+            }
+        }
+    }
+
+    pub async fn delete_objects(&self, bucket: &str, key: &str) -> Result<(), AwsError> {
+        let list_objects_to_delete = self.list_files(bucket, key).await?;
+        let mut delete_objects_builder = Delete::builder();
+
+        let number_of_objects = list_objects_to_delete.len();
+        if number_of_objects == 0 {
+            return Ok(());
+        }
+
+        for obj in list_objects_to_delete {
+            let obj_id = ObjectIdentifier::builder().set_key(Some(obj.key)).build();
+            delete_objects_builder = delete_objects_builder.objects(obj_id);
+        }
+
+        let result = self
+            .client
+            .delete_objects()
+            .bucket(bucket)
+            .delete(delete_objects_builder.build())
+            .send()
+            .await;
+
+        match result {
+            Ok(_) => Ok(()),
+            Err(e) => {
+                let error_message = format!(
+                    "error while s3 delete_objects {}/{}; err={}",
+                    bucket, key, e
+                );
+                tracing::error!(error_message);
+                Err(AwsError::new(error_message.as_str()))
+            }
+        }
+    }
+
+    pub async fn put_object_tag(
+        &self,
+        bucket: &str,
+        key: &str,
+        tag_key: &str,
+        tag_value: &str,
+    ) -> Result<(), AwsError> {
+        let tag = Tag::builder().key(tag_key).value(tag_value).build();
+        let result = self
+            .client
+            .put_object_tagging()
+            .bucket(bucket)
+            .key(key)
+            .tagging(Tagging::builder().tag_set(tag).build())
+            .send()
+            .await;
+
+        match result {
+            Ok(_) => Ok(()),
+            Err(error) => {
+                let error_message = format!(
+                    "error while s3 put_object_tagging {}/{}; err={}",
```

Comment:
Since it's always the function name used in error messages:
```suggestion
                    "error while s3 put_object_tag {}/{}; err={}",
```

Besides, aren't the messages missing a verb? We could use `error while running s3 put_object_tagging' for instance.